bots/tests-invoke

```diff
@@ -83,11 +84,26 @@ class PullTask(object):
         self.sink = None
         self.github_status_data = None
 
-    def start_publishing(self, host):
+    def detect_collisions(self, opts):
         api = github.GitHub()
         if not self.github_revision:
             self.github_revision = self.revision
 
+        if opts.pull_number:
+            pull = api.get("pulls/{0}".format(opts.pull_number))
+            if pull and pull['head']['sha'] != self.revision:
+                return 'Newer revision available on GitHub for this pull request'
+
+        statuses = api.get('commits/' + self.github_revision + '/statuses')
+        for status in statuses:
+            if status.get('context') == self.context and status.get('state') != 'pending':
+                return 'Status of context isn\'t pending'
```

Comment:
It's whimsical, but please use double quotes for strings containing a single quote.